src/tsolvers/TSolverHandler.cc

```diff
@@ -102,5 +102,17 @@ TRes TSolverHandler::check(bool complete)
     return res_final;
 }
 
+// MB: This is currently needed to replace a common array of deduced elements with solver ID
+TSolver* TSolverHandler::getSolverDeducing(PTRef ptref) const {
+    // MB: Can we use solverSchedule? Double check this if theory combination is implemented
+    for (auto* solver : tsolvers) {
+        if (solver != nullptr && solver->isValid(ptref)) {
```

Comment:
Is there some global assertion here that states that ptref is assigned at this point?

Reply:
I will check this.

Reply:
The method has been renamed to better reflect its meaning.